drivers/adc/adc_stm32.c

```diff
@@ -1174,6 +1176,34 @@ static const struct adc_driver_api api_stm32_driver_api = {
 	.ref_internal = STM32_ADC_VREF_MV, /* VREF is usually connected to VDD */
 };
 
+/* st_prescaler property requires 2 elements : clock ASYNC/SYNC and DIV */
+#define STM32_ADC_CLOCK(x, prop) DT_INST_PROP_BY_IDX(x, prop, 0)
+#define STM32_ADC_DIV(x, prop) DT_INST_PROP_BY_IDX(x, prop, 1)
+
+/* macro to check is 1st element is SYNC or ASYNC */
+#define STM32_ADC_CLOCK_IS_SYNC(x, prop) IS_EQ(STM32_ADC_CLOCK(x, prop), SYNC)
+
+/* Macro to set the prefix depending on the 1st element */
+#define STM32_ADC_DO_PREFIX(x, prop)				\
+	COND_CODE_1(IS_EQ(STM32_ADC_CLOCK(x, prop), SYNC),	\
+		prescaler_sync,					\
+		prescaler_async)
+
+/* Concat prefix (1st element) and DIV value (2nd element) of st,prescaler property */
+#if defined(CONFIG_SOC_SERIES_STM32F1X)
+/* not available for that serie */
+#define DT_INST_PRESC_PROP(x, prop) 0
+#else
+/*
+ * The st,prescaler property must declare 2 elements
+ * else choosing the default prescaler clock and div by 4
+ */
+#define DT_INST_PRESC_PROP(x, prop)					\
```

Comment:
I think it is preferred to use _STM32_ namespace as DT_INST_XXX_PROP might be reserved for Zephry generic macros.
```suggestion
#define DT_STM32_PRESC_PROP(x, prop)					\
```

Reply:
ok, changing macro name

Reply:
Also, 'prop' argument should not be required (always the same prop, and is matching the macro name `PRESC` )